File: Example/Daily/2022/05-28/LXCollectionCell.m
//
//  LXCollectionCell.m
//  LXToolKitObjc_Example
//
//  Created by lxthyme on 2022/5/30.
//  Copyright © 2022 lxthyme. All rights reserved.
//

#import "LXCollectionCell.h"

@interface LXCollectionCell() {
}

@end

@implementation LXCollectionCell
#pragma mark -
#pragma mark - 🛠Life Cycle
- (instancetype)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];
    if (self) {
        [self prepareUI];
    }
    return self;
}
- (void)prepareForReuse {
    [super prepareForReuse];
}

#pragma mark -
#pragma mark - 👀Public Actions

#pragma mark -
#pragma mark - 🔐Private Actions

#pragma mark -
#pragma mark - 🍺UI Prepare & Masonry
- (void)prepareUI {
    self.contentView.backgroundColor = [UIColor whiteColor];

    [self.contentView addSubview:self.labTitle];

    [self masonry];
}

#pragma mark Masonry
- (void)masonry {
    [self.labTitle mas_makeConstraints:^(MASConstraintMaker *make) {
        make.edges.equalTo(@0.f);
    }];
}

#pragma mark Lazy Property
- (UILabel *)labTitle {
    if(!_labTitle){
        UILabel *label = [[UILabel alloc]init];
        label.text = @"";
        label.font = [UIFont systemFontOfSize:15.f];
        label.textColor = [UIColor blackColor];
        label.backgroundColor = [UIColor cyanColor];
        label.numberOfLines = 0;
        label.textAlignment = NSTextAlignmentCenter;
        label.lineBreakMode = NSLineBreakByTruncatingTail;
        _labTitle = label;
    }
    return _labTitle;
}
@end

File: LXToolKit_Example/LXToolKit_Example/Classes/Daily/2023/11-09/LXCollectionCell.swift
//
//  LXCollectionCell.swift
//  LXToolKit_Example
//
//  Created by lxthyme on 2023/11/9.
//
import UIKit

class LXCollectionCell: UICollectionViewCell {
    // MARK: 📌UI
    private lazy var labTitle: UILabel = {
        let label = UILabel()
        label.text = ""
        label.font = .systemFont(ofSize: 14)
        label.textColor = .black
        // label.numberOfLines = 0
        label.lineBreakMode = .byTruncatingTail
        label.textAlignment = .center
        label.adjustsFontForContentSizeCategory = true
        label.layer.borderWidth = 0.5
        label.layer.borderColor = UIColor.cyan.cgColor
        label.layer.cornerRadius = 8
        label.layer.masksToBounds = true
        return label
    }()
    required init?(coder aDecoder: NSCoder) { fatalError("init(coder:) has not been implemented") }
    // MARK: 🔗Vaiables
    override init(frame: CGRect) {
        super.init(frame: frame)

        prepareUI()
    }
    open override func prepareForReuse() {
        super.prepareForReuse()
    }
}

// MARK: 🌎LoadData
extension LXCollectionCell {
    public func dataFill(_ title: String) {
        labTitle.text = title
    }
}

// MARK: 👀Public Actions
extension LXCollectionCell {}

// MARK: 🔐Private Actions
private extension LXCollectionCell {}

// MARK: - 🍺UI Prepare & Masonry
private extension LXCollectionCell {
    func prepareUI() {
        self.contentView.backgroundColor = .white

        [labTitle].forEach(self.contentView.addSubview)

        masonry()
    }

    func masonry() {
        labTitle.snp.makeConstraints {
            $0.edges.equalToSuperview()
        }
    }
}
